modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GatewayProtectedCacheProxy.java

```diff
@@ -236,22 +236,26 @@ public void setCacheManager(org.apache.ignite.cache.CacheManager cacheMgr) {
         CacheOperationGate opGate = onEnter();
 
         try {
-            if (context().mvccEnabled()) // Can (should?) be supported in future.
-                throw new UnsupportedOperationException("Read Repair is not supported at MVCC mode.");
+            if (context().mvccEnabled()) {
+                throw new UnsupportedOperationException(
+                    "The TRANSACTIONAL_SNAPSHOT mode does not support the read-repair feature.");
+            }
 
-            if (context().isNear()) // Can be supported in future.
-                throw new UnsupportedOperationException("Read Repair is not supported for near caches.");
+            if (context().isNear())
+                throw new UnsupportedOperationException("Read-repair is not supported by near caches.");
 
-            if (context().readThrough()) // Can be supported in future.
+            if (context().readThrough()) {
                 // Read Repair get operation produces different versions for same entries loaded via readThrough feature.
-                throw new UnsupportedOperationException("Read Repair is not supported for caches with readThrough enabled.");
+                throw new UnsupportedOperationException("Read-repair is not supported by caches with readThrough enabled.");
```

Comment:
Well, in my opinion, the object (IgniteCache) may support/not support a feature (Read-repair) (not a vice-versa).
In general, read-repair is `IgniteCache#get()` operation with additional steps, and it sounds weird to me that read-repair feature does not support #get() op (in case of near caches).
For example, let's consider the following statements:
 - Rolling-upgrade feature does not support Apache Ignite [AI is not supported by RU] hmmm... :) wtf?
 - Apache Ignite does not support rolling-upgrade feature. [RU is not supported by AI] <- sound more natural to me.

Reply:
To be honest, RU implemented inside AI :) 
Anyway, RR is responsible for features support.
RR implementation does not support mvcc. 
MVCC doesn't know anything about RR.
So, correct test is "is not supported by RR feature"

Reply:
> To be honest, RU implemented inside AI :)
It does not matter I think. I would like to concentrate on a feature definition itself.
In my humble opinion, a feature is characteristic of a software item.
RR is an additional value related to cache functionality. It does not make any sense without caches. So, the primary 'item' is cache.

Reply:
You're unable to create withRR proxy because RR feature does not support such cache configuration.
So, you may say "not supported by RR" or "this configuration incompatible with RR feature"

Reply:
And again, it is a part of IgniteCache interface, and you cannot create RR proxy just because this cache does not support RR feature. :)

Reply:
Cache unable to support something.
Operation (RR) incompatible with ... or not allowed for ... cannot be performed on ... 

RR cannot be performed on local cache because it's local :)

Reply:
> Cache unable to support something.

Of course, it able to support something! :) hmm, perhaps, the phrase `Operation (RR) incompatible with...` is good enough.

> RR cannot be performed on local cache because it's local :)

I try to talk about the definition itself and not about the particular case (local cache/near cache)
It seems we need some sort of arbiter :) Maybe, Nikolay Izhikov will step in with his vision about this important discussion )))

Reply:
@sk0x50 
@anton-vinogradov 

My vote for a "Cache configuration(readThrough) incompatible with RR".

Reply:
okay.